### ISSUE TITLE
Regression: cannot select the last option in <select> with ng-options

### ISSUE THREAD

**Issue by BenoitBorlee** [angular/angular.js#10161]
I have a select directive with ng-options on a simple array of strings :
```html
<select ng-model="formData.mySelect" ng-options="o for o in ['A','B','C']"></select>
```
With angularjs 1.2.27, I cannot select the last option ('C') the first time. I can only select other entries.

Reproduced here :
http://jsbin.com/pahujutico/1/watch?html,js,output
Steps :
- Select the 'C' option of the selection box at first.
- Please note that 'C' is NOT displayed.
- Select the 'A' option of the selection box.
- Please note that 'A' is displayed.

It works well with angularjs 1.2.25.

**Comment by petebacondarwin** [comment]
This is also fixed by #10639 - see http://jsbin.com/zejihedovi/1/edit?html,js,output

**Comment by petebacondarwin** [comment]
We are not going to backport this to 1.2.x since that branch is only receiving security fixes now.
